app/uk/gov/hmrc/tariffclassificationfrontend/controllers/CaseController.scala

```diff
@@ -24,37 +24,52 @@ import uk.gov.hmrc.play.bootstrap.controller.FrontendController
 import uk.gov.hmrc.tariffclassificationfrontend.config.AppConfig
 import uk.gov.hmrc.tariffclassificationfrontend.forms.DecisionFormMapper
 import uk.gov.hmrc.tariffclassificationfrontend.models.Case
-import uk.gov.hmrc.tariffclassificationfrontend.service.CasesService
+import uk.gov.hmrc.tariffclassificationfrontend.service.{CasesService, FileStoreService}
 import uk.gov.hmrc.tariffclassificationfrontend.views
 import uk.gov.hmrc.tariffclassificationfrontend.views.CaseDetailPage
 import uk.gov.hmrc.tariffclassificationfrontend.views.CaseDetailPage.CaseDetailPage
 
 import scala.concurrent.Future
+import scala.concurrent.Future.successful
 
 @Singleton
 class CaseController @Inject()(authenticatedAction: AuthenticatedAction,
                                casesService: CasesService,
+                               fileStoreService: FileStoreService,
                                mapper: DecisionFormMapper,
                                val messagesApi: MessagesApi,
                                implicit val appConfig: AppConfig) extends FrontendController with I18nSupport {
 
   def summary(reference: String): Action[AnyContent] = authenticatedAction.async { implicit request =>
-    getCaseAndRenderView(reference, CaseDetailPage.SUMMARY, views.html.partials.case_summary(_))
+    getCaseAndRenderView(reference, CaseDetailPage.SUMMARY, c => successful(views.html.partials.case_summary(c)))
   }
 
   def applicationDetails(reference: String): Action[AnyContent] = authenticatedAction.async { implicit request =>
-    getCaseAndRenderView(reference, CaseDetailPage.APPLICATION_DETAILS, views.html.partials.application_details(_))
+    getCaseAndRenderView(
+      reference,
+      CaseDetailPage.APPLICATION_DETAILS,
+      c => {
+        for {
```

Comment:
I think you should be able to define everything inside a single for comprehension:
```
         for {
          attachments <- fileStoreService.getAttachments(c)
          letter <- fileStoreService.getLetterOfAuthority(c)
          res <- views.html.partials.application_details(c, attachments, letter)
        } yield res
```

Reply:
Will fix